rlclientlib/federation/joined_log_provider.h

```diff
@@ -26,6 +27,11 @@ struct i_joined_log_provider
 {
   virtual ~i_joined_log_provider() = default;
 
+  // Add events to the joiner to be processed when invoke_join() is called.
+  // The contents of the binary data buffer is implementation dependent.
+  RL_ATTR(nodiscard)
+  virtual int add_events(const std::vector<std::shared_ptr<utility::data_buffer>>& events, api_status* status = nullptr) = 0;
```

Comment:
The interface of the log provider should not have the capacity to add logs to it. How the logs are produced should be an implementation detail of the concrete class which implements this interface